src/CrudTrait.php

```diff
@@ -191,4 +191,215 @@ public function uploadMultipleFilesToDisk($value, $attribute_name, $disk, $desti
+        }
+
+        //Needed for the original image
+        if (! array_key_exists('original', $variations)) {
+            $variations['original'] = null;
+        }
+
+        //Needed for admin thumbnails
+        if (! array_key_exists('thumb', $variations)) {
+            $variations['thumb'] = [150, 150];
+        }
+
+        $request = \Request::instance();
+
+        //We need to setup the disk paths as they're handled differently
+        //depending if you need a public path or internal storage
+        $disk_config = config('filesystems.disks.'.$disk);
+        $disk_root = $disk_config['root'];
+
+        //if the disk is public, we need to know the public path
+        if ($disk_config['visibility'] == 'public') {
+            $public_path = str_replace(public_path(), '', $disk_root);
+        } else {
+            $public_path = $disk_root;
+        }
+
+        // if a new file is uploaded, delete the file from the disk
+        if (($request->hasFile($attribute_name) || starts_with($value, 'data:image')) && $this->{$attribute_name}) {
+            foreach ($variations as $variant => $dimensions) {
+                $variant_name = str_replace('-original', '-'.$variant, $this->{$attribute_name});
+                \Storage::disk($disk)->delete($variant_name);
+            }
+            $this->attributes[$attribute_name] = null;
+        }
+
+        // if the file input is empty, delete the file from the disk
+        if (empty($value)) {
+            foreach ($variations as $variant => $dimensions) {
+                $variant_name = str_replace('-original', '-'.$variant, $this->{$attribute_name});
+                \Storage::disk($disk)->delete($variant_name);
+            }
+
+            return $this->attributes[$attribute_name] = null;
+        }
+
+        // if a new file is uploaded, store it on disk and its filename in the database
+        if ($request->hasFile($attribute_name) && $request->file($attribute_name)->isValid()) {
+
+            // 1. Generate a new file name
+            $file = $request->file($attribute_name);
+            $new_file_name = md5($file->getClientOriginalName().time());
+            $new_file = $new_file_name.'.'.$file->getClientOriginalExtension();
+
+            // 2. Move the new file to the correct path
+            $file_path = $file->storeAs($destination_path, $new_file, $disk);
+            $image_variations = [];
+
+            // 3. but only if they have the ability to crop/handle images
+            if (class_exists('\Intervention\Image\ImageManagerStatic')) {
+                foreach ($variations as $variant => $dimensions) {
+                    $img = \Intervention\Image\ImageManagerStatic::make($file);
+
+                    $variant_name = $new_file_name.'-'.$variant.'.'.$file->getClientOriginalExtension();
+                    $variant_file = $destination_path.'/'.$variant_name;
+
+                    if ($dimensions) {
```

Comment:
That works but `isset($dimensions)` is probably more correct? Or even a check to see what the dimensions are - couldn't someone include a truthy dimensions that are just unexpected?